tempus_fugit_minecraft/world.py

```diff
@@ -0,0 +1,243 @@
+        @see [Issue#86](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/86)
+        """
+        tree = []
+
+        # Create trunks
+        for stem in range(trunk_height):
+            position = (x, y + stem, z)
+            tree.append((TREE_TRUNK, position))
+
+        # Create leaves
+        for dx in range(-2,3):
+            for dy in range(0,3):
+                for dz in range(-2,3):
+                    position = (x + dx, y + trunk_height + dy, z + dz)
+                    tree.append((TREE_LEAVES, position))
+        return tree
+
+    @staticmethod
+    def generate_clouds(world_size: int = WORLD_SIZE, num_of_clouds=int((WORLD_SIZE * 3.75))) -> list[list[tuple[Block, Position]]]:
+        """!
+        @brief Generate sky cloud positions.
+        @param world_size Half the world's size.
+        @param num_of_clouds Number of clouds 
+            (default is "WORLD_SIZE * 3.75").
+        @return clouds list of lists representing cloud block 
+            types and positions.
+        @see [Issue#20](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/20)
+        @see [Issue#28](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/28)
+        @see [Issue#44](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/44)
+        @see [Issue#84](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/84)
+        @see [Issue#86](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/86)
+        """
+        game_margin = world_size
+        clouds = list()
+        for _ in xrange(num_of_clouds):
+            cloud_center_x = random.randint(-game_margin, game_margin)
+            cloud_center_z = random.randint(-game_margin, game_margin)
+            cloud_center_y = random.choice([18,20,22,24,26])
+            s = random.randint(3, 6) # 2 * s is the side length of the cloud
+
+            single_cloud = World.generate_single_cloud(cloud_center_x, cloud_center_y, cloud_center_z, s)
+
+            clouds.append(single_cloud)
+        return clouds
+
+    @staticmethod
+    def generate_single_cloud(cloud_center_x, cloud_center_y, cloud_center_z,s) -> list[tuple[Block, Position]]:
+        """!
+        @brief generate a single cloud (list of cloud blocks).
+        
+        @param cloud_center_x Represents the x-coordinate center of the 
+            cloud.
+        @param cloud_center_x Represents the y-coordinate (height) 
+            center of the cloud.
+        @param cloud_center_x Represents the z-coordinate center of the 
+            cloud.
+        @param s represent number of blocks drawn from center - goes in 
+            each direction around the center.
+        
+        @return single_cloud A list that contains pairs of blocks and
+            positions that represent a cloud. 
+        @see [Issue#20](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/20)
+        @see [Issue#28](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/28)
```

Comment:
I believe this function was not created when we solve these two issues ([Issue#20](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/20) , [Issue#28](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/28)). instead I believe the related issues are:
([Issue#84](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/84) , [Issue#86](https://github.com/WSUCEG-7140/Tempus_Fugit_Minecraft/issues/86))

Reply:
You are correct that while this function wasn't created initially to sovle issues 20 and 28, the functions this replaces *was* meant to solve those two issues. This function is now a combination of `place_cloud_blocks`'s randomization of cloud color, and the block generative aspect of `generate_single_cloud`, both of which were originally in model.py.

I tagged this with #20 and #28 to show that this function still solves the problems those issues were created for and show how the code has evolved from when those issues were created to how they were initially fixed and now for this new evolution to solve the same issue.